app/components/markdown/at_mention/at_mention.tsx

```diff
@@ -93,6 +101,47 @@ const AtMention = ({
         return user.mentionKeys;
     }, [currentUserId, mentionKeys, user]);
 
+    // Checks if the mention is a group
+    const group = useMemo(() => {
+        if (user?.username) {
+            return undefined;
+        }
+        const getGroupsByName = (gs: GroupModelType[]) => {
+            const groupsByName: Dictionary<GroupModelType> = {};
+
+            for (const g of gs) {
+                groupsByName[g.name] = g;
+            }
+
+            return groupsByName;
+        };
+
+        const groupsByName = getGroupsByName(groups);
```

Comment:
```suggestion
            const groupsByName = gs.reduce<Dictionary<GroupModelType>>((acc, g) => {
                acc[v.name] = v;
                return acc;
            }, {})
```

0/5, but looks more compact.

Reply:
I think we lean towards taking the performance benefit of the `for ...` loop here over the array util methods, and I also vaguely remember typescript not transpiling reduce to JS appropriately under certain conditions. Not sure if there is a huge performance benefit here though as I doubt groups will be that large of a list; I'm also 0/5 on this.